internal/config/server/events.go

```diff
@@ -56,3 +58,9 @@ type AggregatorConfig struct {
 	// This is the threshold between rule evaluations + actions.
 	LockInterval int64 `mapstructure:"lock_interval" default:"30"`
 }
+
+// ReminderSubscriber is the configuration for the reminder subscriber to connect to the event queue
+type ReminderSubscriber struct {
+	Enabled    bool                  `mapstructure:"enabled" default:"false"`
+	Connection config.DatabaseConfig `mapstructure:"connection" default:"{}"`
```

Comment:
The intention for reminder subscriber was not to populate the defaults with `config.DatabaseConfig` defaults:

```go
// DatabaseConfig is the configuration for the database
type DatabaseConfig struct {
	Host     string `mapstructure:"dbhost" default:"localhost"`
	Port     int    `mapstructure:"dbport" default:"5432"`
	User     string `mapstructure:"dbuser" default:"postgres"`
	Password string `mapstructure:"dbpass" default:"postgres"`
	Name     string `mapstructure:"dbname" default:"minder"`
	SSLMode  string `mapstructure:"sslmode" default:"disable"`
}
```
^^ These are the defaults for the minder server, makes no sense to have these as defaults for reminder subscriber connection.

---

Also, after you mentioned, I checked. This `default:"{}"` would not work as I am stating above. This is because `SetViperStructDefaults` skips `default` tag for structs.

```go
if field.Type.Kind() == reflect.Struct {
    setViperStructDefaults(v, valueName+".", reflect.Zero(field.Type).Interface())
    continue
}
```